### PR TITLE
[SMALLFIX] Merge Packet{In,Out}Stream with Block{In,Out}Stream

### PR DESCRIPTION
**Pull request by peisun1115**

### CODE REVIEW

File: core/client/fs/src/main/java/alluxio/client/block/stream/BlockOutStream.java [hunk 1]
```diff
@@ -53,63 +64,210 @@ public static BlockOutStream create(FileSystemContext context, long blockId, lon
...
+  }
+
+  /**
+   * @return the remaining size of the block
+   */
+  @Override
+  public long remaining() {
+    long pos = Long.MAX_VALUE;
+    for (PacketWriter packetWriter : mPacketWriters) {
+      pos = Math.min(pos, packetWriter.pos());
```
Comment (calvinjia) [inline core/client/fs/src/main/java/alluxio/client/block/stream/BlockOutStream.java]: Should we handle the case where there are no `PacketWriters`?

File: core/client/fs/src/main/java/alluxio/client/block/stream/BlockInStream.java [hunk 2]
```diff
@@ -17,42 +17,51 @@
...
- * stream is closed.
+ * Provides an {@link InputStream} implementation that is based on {@link PacketReader}s to
+ * stream data packet by packet.
  */
 @NotThreadSafe
-public class BlockInStream extends FilterInputStream implements BoundedStream, Seekable,
-    PositionedReadable, Locatable {
-  /** Helper to manage closeables. */
+public class BlockInStream extends InputStream implements BoundedStream, Seekable,
+    PositionedReadable, Locatable{
```
Comment (calvinjia) [inline core/client/fs/src/main/java/alluxio/client/block/stream/BlockInStream.java]: space after `Locatable`

Comment (peisun1115) [inline core/client/fs/src/main/java/alluxio/client/block/stream/BlockInStream.java]: done

File: core/client/fs/src/main/java/alluxio/client/block/stream/BlockOutStream.java (hunk 1 above)
Comment (peisun1115) [inline core/client/fs/src/main/java/alluxio/client/block/stream/BlockOutStream.java]: It should never be empty.

**Comment by peisun1115** [comment]
@calvinjia PTAL. thanks

**Comment by calvinjia** [comment]
Jenkins, test this please.

**Pull request merged**